### PR TITLE
Change the name of the 'Barrier' module to the plural 'Barriers'

### PR DESCRIPTION
**Pull request by daviditen**
Previously, the `Barrier` module and the `Barrier` record shared a name, which
could lead to problems, as discussed in issue #6979. Update the name of the
module to `Barriers`.

One other idea was to change the name of the record to all lowercase `barrier`,
and change the constructor to an `init` function, but then the `barrier`
method looked like a constructor, so would also need to change names.
Updating the module name appears to be a lower impact change.

The `Barrier` module can still be used by its old name, but now generates
a deprecated warning.

Update tests to use the new module name.  Added a test for the deprecated
warning.

Passed a regular Linux64 paratest.

### CODE REVIEW

File: modules/standard/Barrier.chpl [hunk 1]
```diff
@@ -17,349 +17,9 @@
...
-   In the future, we expect to add more language-level support for a
-   "task-team" concept.  A task-team will more directly support collective
-   operations such as barriers between the tasks within a team.
-
-   The current implementation is designed for correctness, but is not expected
-   to perform well at scale.  We expect performance at scale to improve as
-   this barrier implementation is optimized and as the task-team concept is
-   implemented and optimized.
-*/
+/* Deprecated. Use the module :mod:`Barriers` instead. */
```
Comment (ben-albrecht) [inline modules/standard/Barrier.chpl]: Should we remove this from the documentation or do we tend to leave the deprecated module documented for a release?

Comment (daviditen) [inline modules/standard/Barrier.chpl]: IMO if the code is there it should be documented, even if the documentation is as simple as saying "Don't use this - use this other thing instead."

**Comment by daviditen** [comment]
Thanks @ben-albrecht and @vasslitvinov.

File: modules/standard/Barrier.chpl (hunk 1 above)
Comment (mppf) [inline modules/standard/Barrier.chpl]: I think it's reasonable to have a comment here but I don't think we should be putting deprecated modules into the chpldoc'd set & on to the web site. I think that'll just confuse new users.

Comment (ben-albrecht) [inline modules/standard/Barrier.chpl]: Note that we removed the documentation of `LayoutCSR.chpl` from the public docs, so we should resolve this inconsistency in how we document deprecated modules. If this isn't a well-defined protocol (which it doesn't appear to be), then we should get input from the broader team and try to define a protocol.

**Pull request merged**